mlrun/serving/system_steps.py

```diff
@@ -96,15 +98,19 @@ def reconstruct_request_resp_fields(
                     "outputs and inputs are not in the same length check 'input_path' and "
                     "'output_path' was specified if needed"
                 )
-        request = {"inputs": inputs, "id": getattr(event, "id", None)}
-        resp = {"outputs": outputs}
+        request = {
+            "inputs": inputs,
+            "id": getattr(event, "id", None),
+            "input_schema": input_schema or new_input_schema,
+        }
+        resp = {"outputs": outputs, "output_schema": output_schema or new_output_schema}
```

Comment:
Not the opposite
 ```suggestion
        resp = {"outputs": outputs, "output_schema": new_output_schema or output_schema}
```

Same for inputs.

Reply:
I think it better this way allowing the user priority and allow us to fail him and catch bugs